kutt.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/bugsnag/microkit/clog"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+)
+
+type Client struct {
+	HTTPClient *http.Client
+	ApiKey     string
+	BaseURL    string
+	UserAgent  string
+}
+
+type SubmitParams struct {
+	URL      string `json:"target"`
+	ExpireIn string `json:"expire_in"`
+}
+
+func NewLinksClient() *Client {
+	var cli Client
+	cli.ApiKey = viper.GetString("kutt_api_key")
+	cli.BaseURL = viper.GetString("kutt_base_url")
+	cli.UserAgent = "promalert/v1 (+https://github.com/bugsnag/promalert)"
+	cli.HTTPClient = &http.Client{
+		Timeout: time.Second * 10,
+	}
+	return &cli
+}
+
+func (cli *Client) error(statusCode int, body io.Reader) error {
+	buf, err := ioutil.ReadAll(body)
+	if err != nil || len(buf) == 0 {
+		return errors.Errorf("Request failed with status code %d", statusCode)
+	}
+	return errors.Errorf("StatusCode: %d, Error: %s", statusCode, string(buf))
+}
+
+func (cli *Client) do(req *http.Request) (*http.Response, error) {
+	req.Header.Set("X-API-Key", cli.ApiKey)
+	req.Header.Set("User-Agent", cli.UserAgent)
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept", "application/json")
+	return cli.HTTPClient.Do(req)
+}
+
+func (cli *Client) Submit(ctx context.Context, target string) (*URL, error) {
+	reqURL := fmt.Sprintf("%s/%s", cli.BaseURL, "api/url/submit")
```

Comment:
We should be using the v2 api:
```suggestion
	reqURL := fmt.Sprintf("%s/%s", cli.BaseURL, "api/v2/links")
```